### ISSUE TITLE
`Kokkos::realloc` drops the view label

### ISSUE THREAD

**Issue by dalg24** [kokkos/kokkos#5364]
Bug present 3.7 RC but not in 3.6
```C++
  Kokkos::View<int*> v("v", 0);
  Kokkos::realloc(v, 1);
  ASSERT_EQ(v.label(), "v");
```
yields
```
Failure
Expected equality of these values:
  v.label()
    Which is: ""
  "v"
```

**Comment by dalg24** [comment]
Reported by @aprokop 
Caught by an ArborX unit test that unfortunately was not enabled in the nightly builds that test against Kokkos develop.

**Comment by PhilMiller** [comment]
Fix was merged to develop by #5365 and to 3.7 by #5369
